inspirehep/dojson/hep/model.py

```diff
@@ -29,16 +29,28 @@
 
 def add_book_info(record, blob):
     """Add link to the appropriate book record."""
+    collections = []
     if 'collections' in record:
-        collections = [c['primary'].lower() for c in record['collections']
-                       if 'primary' in c]
+        for c in record['collections']:
+            if 'primary' in c:
+                if isinstance(c['primary'], list):
```

Comment:
This would better if avoided here https://github.com/inspirehep/inspire-next/blob/master/inspirehep/dojson/common/base.py#L188 so that you don't need this code.